cura/Settings/MachineManager.py

```diff
@@ -550,17 +550,18 @@ def copyValueToExtruders(self, key: str):
             if extruder_stack != self._active_container_stack and extruder_stack.getProperty(key, "value") != new_value:
                 extruder_stack.userChanges.setProperty(key, "value", new_value)  # TODO: nested property access, should be improved
 
-    ## Copy the value of all settings of the current extruder to all other extruders as well as the global container.
+    ## Copy the value of all manually changed settings of the current extruder to all other extruders.
     @pyqtSlot()
     def copyAllValuesToExtruders(self):
-        for key in self._active_container_stack.userChanges.getAllKeys():
-            new_value = self._active_container_stack.getProperty(key, "value")
-            extruder_stacks = [stack for stack in ExtruderManager.getInstance().getMachineExtruders(self._global_container_stack.getId())]
-
-            # check in which stack the value has to be replaced
-            for extruder_stack in extruder_stacks:
-                if extruder_stack != self._active_container_stack and extruder_stack.getProperty(key, "value") != new_value:
-                    extruder_stack.userChanges.setProperty(key, "value", new_value)  # TODO: nested property access, should be improved
+        extruder_stacks = list(self._global_container_stack.extruders.values())
+        for extruder_stack in extruder_stacks:
+            if extruder_stack != self._active_container_stack:
+                for key in self._active_container_stack.userChanges.getAllKeys():
+                    new_value = self._active_container_stack.getProperty(key, "value")
+
+                    # check if the value has to be replaced
+                    if extruder_stack.getProperty(key, "value") != new_value:
```

Comment:
Hi @guillemavila , I don't know if this is intended, but please be aware that, with this check, if two extruders happen to have the same values, but one is auto-calculated instead of being modified by the user, that value will not be copied. As a result, if later you change something in that extruder, the auto-calculated value may get changed as well, and you may not have the same values across all extruders.

If you want to make sure that all extruder must have the same values, compare with `extruder_stack.userChanges` instead.

Reply:
Yes, well seen! I removed the check so that it forces all user changed values even if they are the same (auto calculated or not)